code/modules/core_implant/cruciform/machinery/cruciformforge.dm

```diff
@@ -0,0 +1,147 @@
+#define LOAD "load"
+#define WORK "work"
+#define DONE "done"
+
+/obj/machinery/neotheology/cruciformforge
+	name = "Cruciform Forge"
+	desc = "This unique piece of technology can be fed biomatter, plasteel, and gold in order to print more cruciforms. Only a select few NeoTheologian architects know how to create these machines. "
+	icon = 'icons/obj/neotheology_machinery.dmi'
+	icon_state = "cruciforge"
+
+	density = TRUE
+	anchored = TRUE
+	layer = 2.8
+
+	var/power_cost = 250
+
+	var/working = FALSE
+	var/start_working
+	var/work_time = 30 SECONDS
+	var/storage_capacity = 50
+	var/list/stored_material = list()
+	var/list/needed_material = list(MATERIAL_BIOMATTER = 10, MATERIAL_PLASTEEL = 5, MATERIAL_GOLD = 2)
+	var/spawn_type = /obj/item/weapon/implant/core_implant/cruciform
+
+	// A vis_contents hack for materials loading animation.
+	var/tmp/obj/effect/flicker_overlay/image_load
+
+/obj/machinery/neotheology/cruciformforge/Initialize()
+	. = ..()
+
+	image_load = new(src)
+
+	for(var/_material in needed_material)
+		stored_material[_material] = rand(1, 10)
+
+/obj/machinery/neotheology/cruciformforge/examine(user)
+	..()
+
+	var/list/matter_count_need = list()
+	for(var/_material in needed_material)
+		matter_count_need += "[needed_material[_material]] [_material]"
+
+	var/list/matter_count = list()
+	for(var/_material in stored_material)
+		matter_count += " [stored_material[_material]] [_material]"
+
+	to_chat(user, SPAN_NOTICE("Need materials: [english_list(matter_count_need)].\nIt contains: [english_list(matter_count)]."))
+
+/obj/machinery/neotheology/cruciformforge/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack))
+		eat(user, I)
+		return
+
+/obj/machinery/neotheology/cruciformforge/proc/produce()
+	for(var/_material in needed_material)
+		stored_material[_material] -= needed_material[_material]
+
+	use_power(power_cost)
+	working = TRUE
+	start_working = world.time
+	flick_anim(WORK)
+	START_PROCESSING(SSobj, src)
+
+/obj/machinery/neotheology/cruciformforge/Process()
+	if(!working)
+		STOP_PROCESSING(SSobj, src)
+		return
+
+	if(world.time >= (start_working + work_time))
+		flick_anim(DONE)
+		new spawn_type(get_turf(src))
+		working = FALSE
+		STOP_PROCESSING(SSobj, src)
+
+/obj/machinery/neotheology/cruciformforge/proc/eat(mob/living/user, obj/item/eating)
+
+	if(!eating && istype(user))
+		eating = user.get_active_hand()
+
+	if(!istype(eating))
+		return FALSE
+
+	if(stat)
```

Comment:
comment: I don't understand why you need this (I don't ask you to change it, I just want to know).

Reply:
If machine is broken it usually have stat == 0

Reply:
So, according to your code when, will the machine only work when it is broken?

Reply:
Sorry, i misspelling. If it has stat != 0 then it's broken or all what in defines been sad:
// Bitflags for machine stat variable.
#define BROKEN   0x1
#define NOPOWER  0x2
#define POWEROFF 0x4  // TBD.
#define MAINT    0x8  // Under maintenance.
#define EMPED    0x10 // Temporary broken by EMP pulse.